botocore/args.py

```diff
@@ -256,24 +303,27 @@ def _should_force_s3_global(self, region_name, s3_config):
         if s3_config and 'us_east_1_regional_endpoint' in s3_config:
             s3_regional_config = s3_config['us_east_1_regional_endpoint']
             self._validate_s3_regional_config(s3_regional_config)
-        return (
-            s3_regional_config == 'legacy' and
-            region_name in ['us-east-1', None]
+        return s3_regional_config == 'legacy' and region_name in (
```

Comment:
We might want to do the region name check first and then just return or something. This  is a little funky.